### PR TITLE
Accept G, GB and GiB suffixes for options

### LINKED ISSUE #4540: The `--mamory` option should support the G suffix.
**Is your feature request related to a problem? Please describe.**
The `--memory` option lacks support for the G (gigabyte) suffix. `--memory=2G` does not work.

**Describe the solution you'd like**
Add support for the ``G` suffix.

### PR DESCRIPTION
**Pull request by Integral-Tech**
Several options currently accept only K, KB, KiB, M, MB and MiB suffixes. This change adds support for G, GB and GiB so users can specify gigabyte values using the same suffix styles.

Closes: #4540

### CODE REVIEW

File: programs/zstdcli.c
Comment (jlee303) [inline programs/zstdcli.c:422]: Could you also add support for G suffix here for consistency across flags?

File: tests/cli-tests/basic/memlimit.sh [hunk 1]
```diff
@@ -35,6 +35,15 @@ rm file.zst
...
 rm file.zst
+println "+ zstd --memory=1G file"
+zstd -q --memory=1G file && die "Should allow numeric parameter with expected suffix"
+rm file.zst
+println "+ zstd --memory=1GB file"
+zstd -q --memory=1GB file && die "Should allow numeric parameter with expected suffix"
+rm file.zst
+println "+ zstd --memory=1GiB file"
+zstd -q --memory=1GiB file && die "Should allow numeric parameter with expected suffix"
+rm file.zst
```
Comment (jlee303) [inline tests/cli-tests/basic/memlimit.sh:62]: Could you add tests to verify overflow handling and boundary cases for G suffix? (e.g. 3G / 4G)

**Review by jlee303 — CHANGES_REQUESTED** [review]
Looking good, just a few suggestions, thanks for putting up a pr 😄

File: tests/cli-tests/basic/memlimit.sh (hunk 1 above)
Comment (jlee303) [reply tests/cli-tests/basic/memlimit.sh:62]: Consider adding a test for 4G to check we get a clear error message instead of unexpected failure

**Review by jlee303 — APPROVED** [review]
lgtm! Left a small suggestion for additional test case :)

**Comment by Integral-Tech** [comment]
@jlee303 Resolved :) Could you please approve the workflows again?

**Comment by jlee303** [comment]
> @jlee303 Resolved :) Could you please approve the workflows again?

I think it's already been approved by @Cyan4973, I don't see a button on my end :)

**Review by Cyan4973 — APPROVED** [review]
